src/platform/site-wide/user-nav/containers/AutoSSO.jsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { connect } from 'react-redux';
+
+import { autoLogin, autoLogout } from 'platform/user/authentication/utilities';
+import { ssoe, ssoeInbound } from 'platform/user/authentication/selectors';
+import { hasSession, hasSessionSSO } from 'platform/user/profile/utilities';
+import { ssoKeepAliveSession } from 'platform/utilities/sso';
+
+class AutoSSO extends React.Component {
+  state = {
+    hasCalledKeepAlive: false,
+  };
+
+  componentDidUpdate() {
+    const { useSSOe, useInboundSSOe } = this.props;
+    const { hasCalledKeepAlive } = this.state;
+
+    if (useSSOe && useInboundSSOe && !hasCalledKeepAlive) {
+      this.checkStatus();
+    }
+  }
+
+  checkStatus = () => {
+    ssoKeepAliveSession().then(() => {
+      if (hasSession() && hasSessionSSO() === 'false') {
```

Comment:
So that you don't have to work with the stringified versions, I would use `JSON.parse` inside of `hasSessionSSO`. `JSON.parse(sessionStorage.get('someBoolean')) => boolean`. That way you could just use `hasSessionSSO() => boolean`. You could also use `async` and `await` here -

```js
checkStatus = async () => {
   await ssoKeepAliveSession();
   // ...
}
```